pilot/pkg/serviceregistry/kube/controller/fake.go

```diff
@@ -219,6 +222,16 @@ func NewFakeControllerWithOptions(t test.Failer, opts FakeControllerOptions) (*F
 	if opts.ServiceHandler != nil {
 		c.AppendServiceHandler(opts.ServiceHandler)
 	}
+
+	t.Cleanup(func() {
+		c.client.Shutdown()
+	})
+	if !opts.SkipRun {
+		t.Cleanup(func() {
```

Comment:
Assuming this is because we want to fail if it's slow, but still wait ~indefinitely to avoid leaks?